pkg/apis/flyteworkflow/v1alpha1/iface.go

```diff
@@ -491,3 +491,14 @@ func GetOutputsFile(outputDir DataReference) DataReference {
 func GetInputsFile(inputDir DataReference) DataReference {
 	return inputDir + "/inputs.pb"
 }
+
+func GetDeckFile(inputDir DataReference, store *storage.DataStore) DataReference {
+	deckURI := inputDir + "/deck.html"
+
+	// Deck file will not exist if we use the older version of flytekit or disable deck
+	metadata, err := store.Head(context.Background(), deckURI)
+	if err == nil && metadata.Exists() {
+		return deckURI
+	}
+	return ""
```

Comment:
Let's not make any network calls here... the checks for the existence of the deck file should happen wherever we populate OutputUri (e.g. [here](https://github.com/flyteorg/flytepropeller/blob/59069d2aee79de353b2e1cc24c1912c89335faab/pkg/controller/nodes/handler/transition_info.go#L53))

Reply:
Updated it